packages/rest/src/__tests__/unit/body-parser.unit.ts

```diff
@@ -90,6 +90,25 @@ describe('body parser', () => {
     });
   });
 
+  it('rejects top-level arrays in application/json', async () => {
+    const req = givenRequest({
+      url: '/',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      payload: [],
+    });
+    const spec = givenOperationWithRequestBody({
+      description: 'data',
+      content: {},
```

Comment:
IMO it's crucial to provide a schema for the request body object, this schema must describe `type: object`.

As I interpret the currently proposed spec, it allows any values for the request body, therefore LoopBack's REST layer must not reject arrays.